packages/schema/lib/schemas/FieldSchema.js

```diff
@@ -137,6 +139,11 @@ module.exports = makeSchema(
         // TODO: Check if it contains one and ONLY ONE '{{input}}'
         pattern: '^.*{{input}}.*$',
       },
+      meta: {
+        description:
+          'Allows for additional metadata to be stored on the field. Supports simple key-values only (no sub-objects or arrays).',
+        $ref: FieldMetaSchema.id,
+      },
```

Comment:
```suggestion
      meta: {
        description:
          'Allows for additional metadata to be stored on the field. Supports simple key-values only (no sub-objects or arrays).',
        $ref: FlatObjectSchema.id,
      },
```

The [/FlatObjectSchema](https://github.com/zapier/zapier-platform/blob/main/packages/schema/docs/build/schema.md#flatobjectschema) does the work without the need for describing a separate `FieldMetaSchema`.

Reply:
Just came across it.

Reply:
Realizing that Fokke suggested we don't include `null` in the valid values, can we make the `FieldMetaSchema` to be similar to `FlatObjectSchema` at least? To have its properties displayed in a table like that of the `FlatObjectSchema`.

![](https://cdn.zappy.app/fa739d9bf84da34bc1acaec848b46980.png)

Reply:
Interesting, I didn't know about the `anyOf` property in combination with `patternProperties`! Here's how it statically validates against a sub-object (same inputField as the one mentioned above):

![Screen Shot 2024-11-20 at 5 58 32 PM](https://github.com/user-attachments/assets/6193dc46-7abe-4e33-a023-f63aba093b6f)

I've updated the PR to use this mechanism! Good call out, thanks!

Reply:
Here's how the Schema docs look now:
![Screen Shot 2024-11-20 at 6 01 54 PM](https://github.com/user-attachments/assets/381e6708-cbaa-452c-86c8-7f5b6396850a)